packages/flutter/test/material/page_test.dart

```diff
@@ -535,97 +535,6 @@ void main() {
     expect(tester.getTopLeft(find.text('Page 2')), const Offset(100.0, 0.0));
   }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS, TargetPlatform.macOS }));
 
-  testWidgets('back gesture while OS changes', (WidgetTester tester) async {
```

Comment:
Sorry, I swear I added a comment about this somewhere but I can't find it. This test concerns the following change in this PR ([code in PageTransitionsTheme](https://github.com/flutter/flutter/pull/141373/files#diff-41923961de924e865545ad350e284d38a886d56134da05aaed3e551b3640a237L768)):

```diff
-TargetPlatform platform = Theme.of(context).platform;
-
-if (route.popGestureInProgress) {
-  platform = TargetPlatform.iOS;
-}
+final TargetPlatform platform = Theme.of(context).platform;
```

So it used to be that we always chose the iOS transition during a pop gesture, because iOS was the only platform that could detect pop gestures. Now Android can detect them, too, so we can't just use this hack.

One thing that we could do is to preserve the TargetPlatform throughout a pop gesture. However, I'd have to remove the `const` from PageTransitionsTheme so I could save the TargetPlatform. Is that worth it?

In the meantime, I've changed the test to explicitly specify the Cupertino transition, which makes it pass. Not sure if that's better than removing it. https://github.com/flutter/flutter/pull/141373/commits/0888ecf6958b00a72f747ef92342b8d92521bee9

Reply:
With that code removed, what happens now when you switch OS while a backswipe is in progress? Does it crash or get stuck somehow? Or does it still behave somewhat reasonable? As long as it behaves reasonable (and ideally we cover that in a test case) I am fine with this change.

Reply:
I have this working how I want it to now in https://github.com/flutter/flutter/pull/141373/commits/6fca2000916cc5e7be542788d22e2fafd6976f8c.

So now when changing OS during a back gesture, the transition that the gesture began with will continue until the end of the gesture. For example if you start in android and switch to iOS in the middle of a gesture, the appearance of the app will change to iOS, but the Android predictive back transition will continue, and your gesture will be tracked, until release. Subsequent back gestures will then use the iOS transition.

This solution also works out of the box with this test, so I've brought the test back with no changes.